tests/unit-tests/test-class-course.php

```diff
@@ -178,7 +178,7 @@ public function testLogInitialPublishDefaultPropertyValues() {
 		$event = $events[0];
 		$this->assertEquals( 0, $event['url_args']['module_count'] );
 		$this->assertEquals( 0, $event['url_args']['lesson_count'] );
-		$this->assertEquals( -1, $event['url_args']['product_id'] );
+		$this->assertEquals( 1, $event['url_args']['product_count'] );
```

Comment:
Hmm. For backwards compatibility purposes, this should be `0`, because the value is `-` (which is set by the metabox when no product is attached). This was covered previously by the call to `intval`, because it would return a falsy value when the product ID was `-`.

So we should probably filter the array of product ID's based on whether they are valid ID numbers. For now, probably it's fine to use `intval` like we did before, although I wonder if we should also filter out IDs that don't belong to existing product in the database? Probably it's not important at this stage, but worth a wee bit of thought.

Reply:
nice! How does array_filtering by `is_numeric` sound?

Reply:
re: product id validation, in a perfect world, wcpc is installed and we would call `get_product_ids` but we can't count on that. I would also hate to duplicate that functionality here.

To start, I will stick to doing the easy one, filtering for non-numerics, as suggested.

Maybe this is something we can solve by adding a filter to the place in the code where we are about to send that event and hook into that from wcpc and "do the right thing" for product id validation.

Reply:
simple case should be addressed in https://github.com/Automattic/sensei/pull/2755/commits/b6012c1d759623adb4b132ef10971e7e5baed968

Reply:
Looks good to me!